File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.Arrays;
import java.util.ArrayList;
class Solution {
    public int[] solution(int[] answers) {
        int[][] pattern = {
            {1,2,3,4,5},
            {2,1,2,3,2,4,2,5},
            {3,3,1,1,2,2,4,4,5,5,} 
        }; //패턴 정리
        
        int[] result = new int[3]; // 3명 동점일수 있으니까
        
        for(int i=0;i< answers.length;i++){
            for(int j=0;j<pattern.length ;j++){
                if(answers[i] == pattern[j][i%pattern[j].length]){
                    result[j]++;
                }
            }
        }
        
        int maxScore = Arrays.stream(result).max().getAsInt();
        
        ArrayList<Integer> answer = new ArrayList<>();
        for(int i=0; i<result.length;i++){
            if(result[i] == maxScore){
                answer.add(i+1);
            }
        }
        return answer.stream().mapToInt(Integer::intValue).toArray();
        
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    first = [1,2,3,4,5]
    second = [2,1,2,3,2,4,2,5]
    third = [3,3,1,1,2,2,4,4,5,5]
    
    fir = 0
    sec = 0
    trd = 0
    
    for i in range(len(answers)):
        if answers[i] == first[i%5]:
            fir = fir + 1
        if answers[i] == second[i%8]:
            sec = sec + 1
        if answers[i] == third[i%10]:
            trd = trd + 1
    
    max_score = max(fir,sec,trd)
    ans_list = []
    ans_list.append(fir)
    ans_list.append(sec)
    ans_list.append(trd)
    
    if max_score ==  fir:
        answer.append(1)
    if max_score == sec:
        answer.append(2)
    if max_score == trd:
        answer.append(3)
            
        
            
    return answer